lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -175,7 +178,13 @@ protected function getType(&$value)
                     }
                 }
 
-                return self::T_IDENTIFIER;
+                if (strpos($value, '\\') !== false) {
+                    return self::T_QUALIFIED_NAME;
+                } else if (strpos($value, ':') !== false) {
```

Comment:
@mpdude no `SomeAlias:Some\Sub\Class` is absolutely a bad idea :-) Anyway, we should really kill aliases at some point.